File: algo_expert/sorting_algorithms/bubbleSort.js

function swap(array, i, j){
    temp = array[i];
    array[i] = array[j];
    array[j] = temp;
}



function bubbleSort(array) {
    let isSorted = false;
    lastIndex = array.length - 1
    while (!isSorted){
        isSorted = true
        for (let i=0; i<lastIndex; i++){
            if(array[i] > array[i+1]){
                swap(array, i, i+1)
                isSorted = false
            }
        }
    }
    return array;
  }


  let undsortedArray = [7 ,3 ,1 ,4 ,2];
  console.log(`Unsorted array ${undsortedArray}`)
  console.log(`Sorted array ${bubbleSort(undsortedArray)}`)

File: algo_expert/sorting_algorithms/bubbleSort.py


def swap(array, i, j):
    array[i], array[j] = array[j], array[i]


def bubbleSort(array):
    # Write your code here.
    for i in range(1, len(array)):
        is_sorted = True
        for j in range(0, len(array) - i):
            if array[j] > array[j+1]:
                swap(array, j, j+1)
                is_sorted = False
        if is_sorted:
            break
            
    return array


input_arr = [4, 54, 1, 0, -4, 9]
print(f"Input array goes as follows {input_arr}")
print(f"Sorted array {bubbleSort(input_arr)}")


input_arr = [2, 1]
print(f"\n\n Input array goes as follows {input_arr}")
print(f"Sorted array {bubbleSort(input_arr)}")